xrmocap/transform/keypoints3d/optim/prior_optimizer.py

```diff
@@ -0,0 +1,188 @@
+    ):
+        """Initialization for MovementConstrain optimizor.
+
+        Args:
+            verbose (bool, optional):
+                Whether to log info. Defaults to True.
+            bbox_size (list, optional):
+                Edge length of the bounding box.
+                Defaults to [2.0, 2.0, 2.0].
+            stand_only (bool, optional):
+                Only apply to the stand position.
+                Defaults to False.
+            ground_norm (list, optional):
+                Norm of the ground in the world coordinate.
+                This is used to approximate whether the person
+                is in the standing position.
+                Defaults to [0.0, 1.0, 0.0].
+            standing_thr (float, optional):
+                Threshold for standing position.
+                Defaults to 0.0.
+            use_limb_length (bool, optional):
+                Whether to use bone length prior constrain
+                between frames.
+                Defaults to False.
+            n_max_frame (int, optional):
+                Max number of frames used to constrain
+                the bone length. Defaults to 5.
+            diff_thr (float, optional):
+                Threshold for the changes of bone length
+                between frames in ratio. Defaults to 0.5.
+            update_mask (bool, optional):
+                Update keypoints mask for those keypoints
+                set to nan in this optimizer. Defaults to False.
+            logger (Union[None, str, logging.Logger], optional):
+                Logger for logging. If None, root logger will be selected.
+                Defaults to None.
+
+        Returns:
+            Keypoints: Optimized keypoints3d
+        """
+        super().__init__(verbose=verbose, logger=logger)
+
+        self.stand_only = stand_only
+        self.ground_norm = np.array(ground_norm)
+        self.standing_thr = standing_thr
+        self.use_limb_length = use_limb_length
+        self.n_max_frame = n_max_frame
+        self.diff_thr = diff_thr
+        self.offset = np.array(bbox_size) / 2.0
+        self.update_mask = update_mask
+
+    def optimize_keypoints3d(self, keypoints3d: Keypoints,
+                             **kwargs: dict) -> Keypoints:
+
+        # get kps3d array
+        keypoints3d_optim = keypoints3d.clone()
+        kps3d = keypoints3d_optim.get_keypoints()[..., :3]
+        kps3d_mask = keypoints3d_optim.get_mask()
+        n_frame, n_max_person, n_kps, _ = kps3d.shape
+
+        kps3d_optim = np.full((n_frame, n_max_person, n_kps, 4), np.nan)
+
+        # get body center from shoulder and hip keypoints
+        shoulder_keypoint_idxs = get_keypoint_idxs_by_part(
+            'shoulder', convention=keypoints3d_optim.get_convention())
+        hip_keypoint_idxs = get_keypoint_idxs_by_part(
+            'hip', convention=keypoints3d_optim.get_convention())
+        hip_shoulder_idxs = shoulder_keypoint_idxs + hip_keypoint_idxs
```

Comment:
`hip_shoulder_idxs` contains not only hips. What about something like `critical_kps_idxs` or `target_kps_idxs`?